### PR TITLE
Core: Serialize SerializationMode, fixes #839

### LINKED ISSUE #839: Property Serialization mode is not remebered for dynamically added properties.
When loading a network with a processor having one or more dynamically added properties with serialization mode set to `All`, the properties recreated by the deserializer does not have the serialization mode set to All. 

For example: The parallel coordinates have a `DataFrameColormapProperty`. The `DataFrameColormapProperty` dynamically adds a bunch of `ColormapProperty` and set them their serialization mode to `All`.   https://github.com/inviwo/inviwo/blob/1f3aba262ce7f3b9a9f5773d08992947c00657b1/modules/dataframe/src/properties/dataframecolormapproperty.cpp#L101-L114

When the network is saved and have been reloaded the serialization mode of the properties are `Default` which means the next time the network is saved information such as display name etc. will be lost. 

**Suggestion:** In `PropertyOwner`, when adding a property to `ownedProperties_` the serialization mode should be set to `All`.   IMO, this makes sense since we need it for properties recreated by the deserializer to get, e.g, displayname correct

### PR DESCRIPTION
**Pull request by r-englund**
Serialize SerializationMode, fixes #839

### CODE REVIEW

**Comment by ResearchDaniel** [comment]
I think we need to add documentation on what the different modes do.
enum class PropertySerializationMode { Default = 0, All, None };

**Review by petersteneteg — CHANGES_REQUESTED** [review]
Why do we need any on change callback for this?

File: include/inviwo/core/properties/property.h [hunk 1]
```diff
@@ -378,7 +378,7 @@ class IVW_CORE_API Property : public PropertyObservable,
     void notifyAboutChange();
 
     CallBackList onChangeCallback_;
-    PropertySerializationMode serializationMode_;
+    ValueWrapper<PropertySerializationMode> serializationMode_;
```
Comment (petersteneteg) [inline include/inviwo/core/properties/property.h:381]: make no sense to put this in a value wrapper

File: include/inviwo/core/processors/compositeprocessor.h [hunk 2]
```diff
@@ -179,6 +179,8 @@ class IVW_CORE_API CompositeProcessor : public Processor,
     virtual void onSetReadOnly(Property* property, bool readonly) override;
     virtual void onSetVisible(Property* property, bool visible) override;
     virtual void onSetUsageMode(Property* property, UsageMode usageMode) override;
+    virtual void onSetSerializationMode(Property* property,
```
Comment (petersteneteg) [inline include/inviwo/core/processors/compositeprocessor.h:182]: why?

File: include/inviwo/core/properties/propertyobserver.h [hunk 3]
```diff
@@ -33,6 +33,7 @@
 #include <inviwo/core/common/inviwocoredefine.h>
 #include <inviwo/core/common/inviwo.h>
 #include <inviwo/core/util/observer.h>
+#include <inviwo/core/properties/valuewrapper.h>
```
Comment (petersteneteg) [inline include/inviwo/core/properties/propertyobserver.h:36]: why

File: src/core/processors/compositeprocessor.cpp [hunk 4]
```diff
@@ -319,4 +319,11 @@ void CompositeProcessor::onSetUsageMode(Property* orgProp, UsageMode) {
     }
 }
 
+void CompositeProcessor::onSetSerializationMode(Property* orgProp,
+                                                PropertySerializationMode serializationMode) {
+    if (auto superProperty = getSuperProperty(orgProp)) {
```
Comment (petersteneteg) [inline src/core/processors/compositeprocessor.cpp:324]: why? how will this interact with line 274? that is the reasoning here?

**Review by petersteneteg — CHANGES_REQUESTED** [review]

**Comment by r-englund** [comment]
I'll reopen this when I get the time

**Pull request closed without merging**